commands/dev.ts

```
@@ -52,6 +53,32 @@ export async function dev(options: DevOptions, filename: string) {
 
   await unmountUI();
 
+  const buffer = new Uint8Array(1);
+
+  const keyPressDetectionLoop = async () => {
+    while (true) {
+      try {
+        const n = await Deno.stdin.read(buffer);
+        if (n === null) {
+          console.log("EOF detected, continuing to listen for events...");
+          continue; // Continue listening instead of breaking
+        }
+        const key = String.fromCharCode(buffer[0]);
+        if (key === "r") {
+          await replayLastEvent(deployment);
+        }
+      } catch (e) {
+        if (!(e instanceof Deno.errors.Interrupted)) {
```

whats the purpose of this catch? should we just let errors bubble up?